pmm/pmm-submodules.groovy

```diff
@@ -0,0 +1,139 @@
+                sh '''
+                    set -o errexit
+                    if [ -s ci.yml ]; then
+                        sudo rm -rf results tmp || :
+                        git reset --hard
+                        git clean -fdx
+                        python3 ci.py
+                        . ./.git-sources
+                        echo $pmm_commit > apiCommitSha
+                        echo $pmm_branch > apiBranch
+                        echo $pmm_url > apiURL
+                        echo $pmm_qa_branch > pmmQABranch
+                        echo $pmm_qa_commit > pmmQACommitSha
+                        echo $pmm_ui_tests_branch > pmmUITestBranch
+                        echo $pmm_ui_tests_commit > pmmUITestsCommitSha
+                    else
+                        sudo rm -rf results tmp || :
+                        git reset --hard
+                        git clean -fdx
+                        git submodule foreach --recursive git reset --hard
+                        git submodule foreach --recursive git clean -fdx
+                        git submodule status
+                        export commit_sha=$(git submodule status | grep 'pmm-managed' | awk -F ' ' '{print $1}')
+                        export api_tests_commit_sha=$(git submodule status | grep 'sources/pmm/src' | awk -F ' ' '{print $1}')
+                        export api_tests_branch=$(git config -f .gitmodules submodule.pmm.branch)
+                        export api_tests_url=$(git config -f .gitmodules submodule.pmm.url)
+                        echo $api_tests_commit_sha > apiCommitSha
+                        echo $api_tests_branch > apiBranch
+                        echo $api_tests_url > apiURL
+                        cat apiBranch
+                        cat apiURL
+                        export pmm_qa_commit_sha=$(git submodule status | grep 'pmm-qa' | awk -F ' ' '{print $1}')
+                        export pmm_qa_branch=$(git config -f .gitmodules submodule.pmm-qa.branch)
+                        echo $pmm_qa_branch > pmmQABranch
+                        echo $pmm_qa_commit_sha > pmmQACommitSha
+                        export pmm_ui_tests_commit_sha=$(git submodule status | grep 'pmm-ui-tests' | awk -F ' ' '{print $1}')
+                        export pmm_ui_tests_branch=$(git config -f .gitmodules submodule.pmm-ui-tests.branch)
+                        echo $pmm_ui_tests_branch > pmmUITestBranch
+                        echo $pmm_ui_tests_commit_sha > pmmUITestsCommitSha
+                    fi
+                    export fb_commit_sha=$(git rev-parse HEAD)
+                    echo $fb_commit_sha > fbCommitSha
+                    echo $CHANGE_URL > changeUrl
+                    echo $CHANGE_ID > changeId
+                '''
+                }
+                script {
+                    env.PMM_VERSION = sh(returnStdout: true, script: "cat VERSION").trim()
+                }
```

Comment:
This is the only thing we need to do in this step. The rest - what's above and below - is a waste of compute.

Reply:
We need the files  fbCommitSha and other variables defined in this section in the later stages of the submodules script, hence decided to keep this section

Reply:
Not quite so, since the two other pipelines that this one calls take care of generating all variables that they need apart from those we pass to them explicitly. This is a huge block of redundant code that does not bring value to anybody.

All we need from this main pipeline is to check the VERSION file and, depending on its value, call either of the two. Why would it generate so many variables?

Reply:
Sure, I shall remove this section of the code and check while testing these jobs.

Reply:
I removed most of the steps here except the steps to fetch pmm_branch, since we need to pass this as a parameter to the pmm2/pmm3 submodules job.